android-xbuild-target/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "android-xbuild-target"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate_type = ["cdylib"]
+
+[dependencies.winit]
+version = "0.29.1-beta"
+path = ".."
+features = ["android-game-activity"]
```

Comment:
I think we always tested `native` one on the CI?

Reply:
Testing `native` should prevent issues with having to compile native code :+1:

Reply:
I think we changed that to `native` and it still failed, but we figured in `#winit` with @notgull what is going on. `xbuild` just doesn't pull the platform and we miss the `libandroid` and `liblog` on the CI.

Reply:
You can look into https://github.com/rust-windowing/winit/pull/3149 and I got further with xbuild errors.

Reply:
The `xbuild` design intends to work more-or-less out of the box (when `clang` is installed), so it's quite strange to see this.

Reply:
Well, the main issue was that it was passing `ld` linker, which is a `ld.bfd` which can't cross-link, unlike `lld`. And then we had issues with finding `libandroid`, because I think it used the system clang & lld and they don't know about libraries which `xbuild` could bring.

Reply:
e.g. https://github.com/rust-mobile/xbuild/issues/148

Reply:
> And then we had issues with finding `libandroid`, because I think it used the system clang & lld and they don't know about libraries which `xbuild` could bring.

Might that be a `rustflags` precedence issue?

Reply:
Could be, but I'm not familiar with `xbuild`. I think we tried without `RUSTFLAGS` at all initially. You can open PR based on the one I've linked and try to fix CI if you're familiar with how `xbuild` (I'm not, I can just decide based on what I see in the logs).